src/topp/OpenSwathWorkflow.cpp

```diff
@@ -786,13 +782,62 @@ class TOPPOpenSwathWorkflow
     OPENMS_LOG_INFO << "Loaded " << transition_exp.getProteins().size() << " proteins, " <<
       transition_exp.getCompounds().size() << " compounds with " << transition_exp.getTransitions().size() << " transitions." << std::endl;
 
-    if (tr_type == FileTypes::PQP)
+    if (out_features_type == FileTypes::OSW)
     {
-      if (!out_osw.empty())
-      { // copy the PQP file and name it OSW file
-        std::ifstream  src(tr_file.c_str(), std::ios::binary);
-        std::ofstream  dst(out_osw.c_str(), std::ios::binary | std::ios::trunc);
-        dst << src.rdbuf();
+      if (tr_type == FileTypes::PQP)
+      {
+         // copy the PQP file and name it OSW file
+          std::ifstream  src(tr_file.c_str(), std::ios::binary);
+          std::ofstream  dst(out_features.c_str(), std::ios::binary | std::ios::trunc);
+          dst << src.rdbuf();
+      }
+      else if (tr_type == FileTypes::TSV)
+      {
+        // Convert TSV to .PQP 
+        TransitionTSVFile tsv_reader;
+        TargetedExperiment transition_exp_heavy;
+        tsv_reader.setParameters(tsv_reader_param);
+        tsv_reader.convertTSVToTargetedExperiment(tr_file.c_str(), tr_type, transition_exp_heavy);
+        TransitionPQPFile().convertTargetedExperimentToPQP(out_features.c_str(), transition_exp_heavy);
+
+        // instead of reloading - edit the already loaded transition_exp to be compatible with .pqp format
+        // read the PQP to traMLID mapping
+        auto precursor_traml_to_pqp = TransitionPQPFile().getPQPIDToTraMLIDMap(out_features.c_str(), "PRECURSOR");
+        auto transition_traml_to_pqp = TransitionPQPFile().getPQPIDToTraMLIDMap(out_features.c_str(), "TRANSITION");
+
+        // convert tramlID in transitionExp to PQP ID
+        for (auto & prec : transition_exp.getCompounds())
+        {
+          auto id = precursor_traml_to_pqp.find(prec.id);
+          if (id != precursor_traml_to_pqp.end())
+          {
+            prec.id = id->second;
+          }
```

Comment:
minor improvement because of more narrow scope 
```suggestion          
          if (auto id = precursor_traml_to_pqp.find(prec.id); id != precursor_traml_to_pqp.end())
          {
            prec.id = id->second;
          }
```